File: WordToDigit/wtd.rb
# word to digit soln in rb for codeeval by steven a dunn

def convert(word)
    case word
    when "one" then 1
    when "two" then 2
    when "three" then 3
    when "four" then 4
    when "five" then 5
    when "six" then 6
    when "seven" then 7
    when "eight" then 8
    when "nine" then 9
    when "zero" then 0
    end
end

File.open(ARGV[0], 'r').each_line do |line|
    line = line.chomp.split(';')
    line.each do |word|
        print convert(word)
    end
    puts
end

File: WordToDigit/c/wtd.c
// word to digit soln in c for code eval by steven a dunn

#include <stdio.h>
#include <string.h>

int convert(char*);

int main (int argc, char* const argv[])
{
    FILE *fp = fopen(argv[1], "r");
    if (fp != NULL)
    {
        char line[512];
        while (fgets(line, sizeof(line), fp))
        {
            for (int i = 0; i < sizeof(line); ++i)
                if (line[i] == '\n')
                    line[i] = '\0';

            char* token = strtok(line, ";");
            while (token != NULL)
            {
                printf("%d",convert(token));
                token = strtok(NULL, ";");
            }
            puts("");
        }
        fclose(fp);
    }
    return 0;
}

int convert(char* text)
{
    if (!strcmp(text, "zero"))
        return 0;
    else if (!strcmp(text, "one"))
        return 1;
    else if (!strcmp(text, "two"))
        return 2;
    else if (!strcmp(text, "three"))
        return 3;
    else if (!strcmp(text, "four"))
        return 4;
    else if (!strcmp(text, "five"))
        return 5;
    else if (!strcmp(text, "six"))
        return 6;
    else if (!strcmp(text, "seven"))
        return 7;
    else if (!strcmp(text, "eight"))
        return 8;
    else if (!strcmp(text, "nine"))
        return 9;
    else
        return -1;
}